### PR TITLE
Fixes centcom pod launcher array skipping in preset deletion code

### LINKED ISSUE #58207: TGUI: CentcomPodLauncher preset deletion has improper behavior
## Round ID: no

Removing an array item without adjusting the loop index `i` causes the subsequent array item to be skipped.
https://github.com/tgstation/tgstation/blob/77a052876063ef1e4139319458e8e54a2c426c81/tgui/packages/tgui/interfaces/CentcomPodLauncher.js#L793-L796

You need to add a `--i` after.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

fixes #58207 , and click it for more details since it's a really well written issue

## Changelog
:cl:
fix: Fixes centcom pod launcher array skipping in preset deletion code
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
Discussed on discord, this is a patch over a large issue, fix it properly

File: tgui/packages/tgui/interfaces/CentcomPodLauncher.js [hunk 1]
```diff
@@ -793,6 +793,7 @@ class PresetsPage extends Component {
     for (let i = 0; i < presets.length; i++) {
       if (presets[i].id === deleteID) {
         presets.splice(i, 1);
+        i--;
```
Comment (AnturK) [inline tgui/packages/tgui/interfaces/CentcomPodLauncher.js:796]: Can't you just `presets = presets.filter(preset => preset.id !== deleteID)` here ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: tgui/packages/tgui/interfaces/CentcomPodLauncher.js (hunk 1 above)
Comment (stylemistake) [reply tgui/packages/tgui/interfaces/CentcomPodLauncher.js]: I've committed it, @tralezab please re-test.

**Comment by tralezab** [comment]
so from my testing... i still think there's something wrong? It wasn't deleting 😳

**Comment by tralezab** [comment]
And to be clear I was trying to delete a preset, correct me if that's not what this is for

**Comment by tralezab** [comment]
honestly someone who knows the context of the files (like doombringer) should do this fix

**Pull request closed without merging**